vector/src/main/java/im/vector/app/features/webview/WebChromeEventListener.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.webview
+
+import android.webkit.PermissionRequest
+
+interface WebChromeEventListener {
+
+    /**
+     * Triggered when the web view requests permissions.
+     *
+     * @param request The permission request.
+     */
+    fun onPermissionRequest(request: PermissionRequest) {
```

Comment:
I actually mimicked this from `WebViewEventListener` which is a sibling interface. I'm not sure what the specific reason was there though.

Reply:
Yes please remove the //NO-OP for this as you have only one method. Also you could use a  fun interface, like that: 
```
fun interface ShieldComputer {
      fun onPermissionRequest(request: PermissionRequest)
}
```

Reply:
Ok, made the change. I decided not to make it a fun interface though since we might in future want to extend it by adding further `WebChromeClient` methods.